src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -132,23 +160,25 @@ private bool ShouldTriggerInArgumentLists(Workspace workspace, OptionSet options
             return _shouldTriggerCompletionInArgumentListsExperiment.Value;
         }
 
-        private static async Task<bool?> IsTriggerOnDotAsync(Document document, int characterPosition, CancellationToken cancellationToken)
+        protected override bool IsTriggerOnDot(SyntaxToken token, int characterPosition)
         {
-            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-            if (text[characterPosition] != '.')
-            {
-                return null;
-            }
+            if (!IsDot(token, characterPosition))
+                return false;
 
             // don't want to trigger after a number.  All other cases after dot are ok.
-            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(characterPosition);
+            return token.GetPreviousToken().Kind() != SyntaxKind.NumericLiteralToken;
+        }
+
+        private static bool IsDot(SyntaxToken token, int characterPosition)
+        {
             if (token.Kind() == SyntaxKind.DotToken)
-            {
-                token = token.GetPreviousToken();
-            }
+                return true;
+
+            // if we're right after the first dot in .. then that's considered completion on dot.
+            if (token.Kind() == SyntaxKind.DotDotToken && token.SpanStart == characterPosition)
+                return true;
 
-            return token.Kind() != SyntaxKind.NumericLiteralToken;
+            return false;
```

Comment:
the subclasses only have to answer questions about the dot token.  shared logic for vb/c# moved up.